style-packages.md

```diff
@@ -0,0 +1,255 @@
++++
+date        = "2017-01-14T11:27:27-04:00"
+title       = "Style guideline for Go packages"
+description = "Best practices to organize and name Go packages"
+tags        = [ "readability" ]
+type        = "archived"
+draft       = false
++++
+
+Go is about naming and organization as much as everything else in the language.
+Go code lives in a package, a package is the entry point to access Go code. Understanding
+and establishing good practices around packages is important to author effective Go code.
+
+----
+
+## Organization
+
+Let's begin with suggestions how you should organize Go code and explain conventions about
+locating Go packages. Well-organized Go code is easy to discover,
+use and read. Well-organized code is as critical as well designed APIs. The location, name,
+and the structure of your packages are the first elements your users see and interact with.
+
+### Organize by responsbility
+
+A common baggage from other languages is to organize types together in a package
+called models or types. In Go, we organize code by their functional responsbilities.
+
+``` wrong
+package models // DON'T DO IT!!!
+
+// User represents a user in the system.
+type User struct {...}
+```
+
+Rather than creating a models package and declare all entity types there,
+a User type should live in a user package that provides everything related to users.
+
+```
+package user
+
+// User represents a user in the system.
+type User struct {...}
+
+func QueryUsers(ctx context.Context, q *Query) ([]*User, *Iterator, error)
+```
+
+### Use multiple files
+
+If you have a large package, feel free to use many files.
+For example, an HTTP package might have been logically separated into different files
+according to the HTTP aspect the file handles.
+In the following example, an HTTP package is broken down into a few files:
+header types and code, cookie types and code, the actual HTTP implementation, and
+documentation of the package.
+
+```
+- doc.go       // package documentation
+- headers.go   // HTTP headers types and code
+- cookies.go   // HTTP cookies types and code
+- http.go      // HTTP client implementation, request and response types, etc.
+```
+
+### Keep types close
+
+As a rule of thumb, keep types closer to where they are used. This makes it easy for
+any maintainer (not just the original author) to find a type.
+A good place for a Header struct type might be in `headers.go`.
+
+```
+$ cat headers.go
+
+// Header represents an HTTP header.
+type Header struct {...}
+```
+
+### Use cmd directory for commands
```

Comment:
I am thinking about leaving this out for now, we chatted about it but I don't think it is a high priority item to be on this doc.